content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract.md

```diff
@@ -12,25 +12,25 @@ tags:
 title: Write contracts
 ---
 
-Contracts in the Corda 5 Developer Preview work in the same way as they did in Corda 4. In the context of a CorDapp, contracts define rules that are used to verify transaction inputs and outputs. A CorDapp can have more than one contract, and each contract defines rules for one or more states. The goal of a contract is to ensure that input and output states in transactions are valid and to prevent invalid transactions. All parties wishing to transact in a network must run the same contract(s) for any transaction they’re a party to, verifying that the transaction is valid.
+[Contracts](../../../../../../en/platform/corda/5.0-dev-preview-1/cordapps/key-concepts/key-concepts-contracts.md) define the rules of how states can evolve. A CorDapp can have more than one contract, and each contract defines rules for one or more states. The contract governs the execution of [transactions](../../../../../../en/platform/corda/5.0-dev-preview-1/cordapps/key-concepts/key-concepts-transactions.html) that update the ledger by marking zero or more existing ledger states as historic (the inputs), and producing zero or more new ledger states (the outputs). The contract(s) ensures that input and output states in transactions are valid and prevents invalid transactions from occurring. All parties that wish to transact in a network must run the same contract(s) for any transaction they’re a party to, verifying that the transaction is valid.
 
-Contract files implement the contract interface, containing the `verify` method. The `verify` method takes transactions as input and evaluates them against rules defined as a `requireThat` element. Contract execution is deterministic, and transaction acceptance is based on the transaction’s contents alone.
+Contracts are classes that implement the Contract interface. They must override the `verify` method. The `verify` method takes transactions as input and evaluates them against rules defined as a `requireThat` element. Contract execution must be deterministic, and transaction acceptance is based on the transaction’s contents alone. Contract verification is the final step that governs the evolution of the ledger, thus it can only address on-ledger facts.
```

Comment:
```suggestion
Contracts are classes that implement the `Contract` interface. They must override the `verify` method. The `verify` method takes transactions as input and evaluates them against rules defined as a `requireThat` element. Contract execution must be deterministic, and transaction acceptance is based on the transaction’s contents alone. Contract verification is the final step that governs the evolution of the ledger, thus it can only address on-ledger facts.
```